### DISCUSSION TITLE
using JSX together with `view`?

### DISCUSSION THREAD

**Discussion by joshuahhh** [observablehq/framework#1717] [Q&A]
Is it possible to use JSX output with `view`, so that the React component can return a value to be used as part of reactive dataflow?

I imagine it would look something like:
```jsx
const myValue = view(<MySlider/>)
```

But I'm not sure if it is possible to expose a value from `<MySlider/>`.

An alternative approach is to make a Mutable and pass it into the component, but that takes some more work from the point of view of `<MySlider/>`'s consumer, so I'd rather avoid it if possible.

Thanks!

**Comment by Fil** [comment]
Here's the recommended approach:
https://github.com/observablehq/framework/issues/1503#issuecomment-2218673447

The issue is still open as we need to document this better.
